apps/vessel-history/feature/download-activity/download-activity.hook.ts

```diff
@@ -0,0 +1,147 @@
+              end,
+              color,
+              colorLabels,
+              description,
+              descriptionGeneric,
+              regionDescription,
+              durationDescription,
+              duration,
+              vessel,
+              ...rest
+            }) => ({
+              timestamp: getUTCDateTime(timestamp).toISO(),
+              type,
+              subType: subEvent,
+              latitude: position.lat,
+              longitude: position.lon,
+              start: getUTCDateTime(start).toISO(),
+              end: getUTCDateTime(end).toISO(),
+              description,
+              regionDescription,
+              durationDescription,
+              duration,
+              vessel,
+              ...rest,
+              eventId: id,
+            })
+          )
+        const { parse, transforms } = CSVParser.current
+        const csv = parse(data, {
+          transforms: [transforms.flatten({ objects: true, arrays: true })],
+        })
+        FileSaver(new Blob([csv], { type: 'text/plain;charset=utf-8' }), filename)
+      } catch (e) {
+        console.warn(e)
+      }
+      setDownloadingStatus(false)
+    },
+    []
+  )
+
+  const downloadAllEvents = useCallback(async () => {
+    const downloadDate = DateTime.now().toFormat('yyyyLLddHHmm')
+    const fileName =
+      `vessel-viewer-activity_${vessel.shipname}_${vessel.imo}` +
+      `_${downloadDate}` +
+      `_all activity` +
+      `.csv`
+    downloadEvents(allEvents, fileName)
+  }, [allEvents, downloadEvents, vessel.imo, vessel.shipname])
+
+  const downloadFilteredEvents = useCallback(async () => {
+    const downloadDate = DateTime.now().toFormat('yyyyLLddHHmm')
+    const dateRange =
+      filters.end && filters.start
+        ? `_${filters.start}_${filters.end}`
+        : filters.start
+        ? `_since_${filters.start}`
+        : filters.end
+        ? `_until_${filters.end}`
+        : ''
+    const currentFilters: string =
+      (filtersUpdated as string[]).filter((filter) => !['start', 'end'].includes(filter)).length > 0
+        ? '_' +
+          (Object.keys(filters) as (keyof Filters)[])
+            // Exclude filters without value or false
+            .filter((key) => !!filters[key] && !['start', 'end'].includes(key as string))
+            .map((filter) =>
+              (t(`settings.${filter}.shortTitle` as any, filter) as string).toLocaleLowerCase()
+            )
+            .join(',')
+        : ''
+    const fileName =
+      `vessel-viewer-activity_${vessel.shipname}_${vessel.imo}` +
+      `_${downloadDate}` +
+      `_filtered activity` +
+      `${dateRange}` +
+      `${currentFilters}` +
+      `.csv`
+
+    downloadEvents(eventsList, fileName)
+  }, [downloadEvents, eventsList, filters, filtersUpdated, t, vessel.imo, vessel.shipname])
+
+  // TODO define approach to display readme file
+  const readmeUrl = useMemo(
+    () => `${BASE_URL}/readme.csv.activity.${i18n.language}.pdf`,
```

Comment:
Not sure if it's going to be translated at this point, if not it could just be the same english readme copied to the `readme.csv.activity.fr.pdf` file. I wouldn't put more logic for this. WDYT?